index.js

```diff
@@ -36,22 +42,18 @@ function rfdc (opts) {
 
   function clone (o) {
     if (typeof o !== 'object' || o === null) return o
-    if (o instanceof Date) return new Date(o)
     if (Array.isArray(o)) return cloneArray(o, clone)
-    if (o instanceof Map) return new Map(cloneArray(Array.from(o), clone))
-    if (o instanceof Set) return new Set(cloneArray(Array.from(o), clone))
+    if (constructorHandlers.has(o.constructor)) {
+      return constructorHandlers.get(o.constructor)(o, clone)
+    }
     const o2 = {}
     for (const k in o) {
       if (Object.hasOwnProperty.call(o, k) === false) continue
       const cur = o[k]
       if (typeof cur !== 'object' || cur === null) {
         o2[k] = cur
-      } else if (cur instanceof Date) {
-        o2[k] = new Date(cur)
-      } else if (cur instanceof Map) {
-        o2[k] = new Map(cloneArray(Array.from(cur), clone))
-      } else if (cur instanceof Set) {
-        o2[k] = new Set(cloneArray(Array.from(cur), clone))
+      } else if (constructorHandlers.has(cur.constructor)) {
```

Comment:
```suggestion
      } else if (cur.constructor !== Object && constructorHandlers.has(cur.constructor)) {
```

I think we add a fast path wherever there is a `.has()` to bypass the map lookup.

This should allow to avoid the perf regression.

Reply:
Have pushed the change. It improves performance, but not back to where it was before:

```
benchRfdc*100: 212.82ms
benchRfdcProto*100: 229.3ms
benchRfdcCircles*100: 255.588ms
benchRfdcCirclesProto*100: 238.306ms
```

Reply:
Okay I've assigned in the `if` check to avoid calling both `.has()` and `.get()`, and benchmarks are improved to:

```
benchRfdc*100: 203.999ms
benchRfdcProto*100: 215.779ms
benchRfdcCircles*100: 224.12ms
benchRfdcCirclesProto*100: 243.172ms
```

Which is vaguely in-line with before.